compiler/src/error.rs

```diff
@@ -41,6 +42,24 @@ pub enum CompileErrorType {
 }
 
 impl CompileError {
+    pub fn is_indentation_error(&self) -> bool {
+        if let CompileErrorType::Parse(parse) = &self.error {
+            if let ParseErrorType::Lexical(lex) = parse {
```

Comment:
Instead of `if let` use a match statement here to match the variants of `ParseErrorType`.